src/plugins/intel_gpu/src/graph/graph_optimizer/eltwise_remove_stride.cpp

```diff
@@ -37,12 +37,12 @@ void eltwise_remove_stride::conv_stride_extend(program& p, program_node& node, c
         c->with_output_size = false;
         node.recalc_output_layout(true);
     } else {
-        bool can_shrink_x = (filter_size.spatial[0] - (conv->stride.spatial[0] + (tensor.spatial[0] - 1))) >= 0;
-        bool can_shrink_y = (filter_size.spatial[1] - (conv->stride.spatial[1] + (tensor.spatial[1] - 1))) >= 0;
+        bool can_shrink_x = (filter_size.spatial[0] - (conv->stride[1] + (tensor.spatial[0] - 1))) >= 0;
```

Comment:
It's more about existing logic, but anyway... It looks like we can process any format with i8 or f32 data types here, because of (in eltwise_remove_stride::run):
```
if (node->get_output_layout().data_type != data_types::i8 &&
    node->get_output_layout().data_type != data_types::f32)
...
continue
```
Maybe it makes sense to extend this logic for 3d cases or prohibit any formats with spatial_rank != 2

Reply:
Added rank check